File: Swu.Portal.Data/Models/News.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Swu.Portal.Data.Models
{
    public class News :IEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Title_TH { get; set; }
        public string Title_EN { get; set; }
        public string ImageUrl { get; set; }
        public string CreatedBy { get; set; }
        public string FullDescription_TH { get; set; }
        public string FullDescription_EN { get; set; }
        public DateTime StartDate { get; set; }
        [DefaultValue(false)]
        public bool IsActive { get; set; }
    }
}

File: Swu.Portal.Web/Scripts/model/news.ts
module Swu {
    export interface INews {
        id: number;
        imageUrl?: string;
        title?: string;
        description?: string;
        createdBy: string;
        startDate: Date;

        title_en?: string;
        description_en?: string;

        title_th?: string;
        description_th?: string;
    }
}